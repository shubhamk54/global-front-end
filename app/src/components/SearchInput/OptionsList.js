import PropTypes from 'prop-types';
import React from 'react';
import './OptionsList.scss';


function OptionsList({ ...props }) {
  const { options, selectedOptionIndex, onClick } = props;

  return (
    <ul className="suggestions">
      {options.map((option, optionIndex) => (
        <li
          className={optionIndex === selectedOptionIndex ? 'suggestion-active' : ''}
          key={option}
          onClick={() => onClick(option)}
        >
          {option}
        </li>
      ))}
    </ul>
  );
}

OptionsList.propTypes = {
  options: PropTypes.array.isRequired,
  selectedOptionIndex: PropTypes.number,
  onClick: PropTypes.func,
};


export default OptionsList;
