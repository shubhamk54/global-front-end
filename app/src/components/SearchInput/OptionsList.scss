.suggestions {
  position: absolute;
  z-index: 2;
  background: white;
  border: 1px solid #999;
  border-top-width: 0;
  list-style: none;
  margin-top: 1px;
  overflow-y: auto;
  padding-left: 0;
  min-width: -webkit-fill-available;
  box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
  border-radius: 3px 3px 14px 14px;
}

.suggestions li {
  padding: 0.5rem;
  text-decoration: none;
  display: block;
}

.suggestion-active,
.suggestions li:hover {
  background-color: #ddd;
  cursor: pointer;
}

.suggestions li:not(:last-of-type) {
  border-bottom: 1px solid #999;
}
