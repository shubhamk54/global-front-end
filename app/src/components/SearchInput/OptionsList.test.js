import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import OptionsList from './OptionsList.js';

const fakeProps = {
  options: ['test-1', 'test-2', 'test-3'],
  selectedOptionIndex: 1,
  onClick: () => jest.fn(),
};

describe('Snapshot testing for OptionsList component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<OptionsList {...fakeProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Chldern rendering tests for OptionsList component', () => {
  it('OptionsList renders single ul tag', () => {
    const wrapper = mount(<OptionsList {...fakeProps} />);

    expect(wrapper.find('ul').children()).toHaveLength(fakeProps.options.length);
  });

  it('OptionsList text on click event', () => {
    const mockCallBack = jest.fn();

    const wrapper = mount(
      <OptionsList
        options={fakeProps.options}
        selectedOptionIndex={fakeProps.selectedOptionIndex}
        onClick={mockCallBack}
      />,
    );
    const lists = wrapper.find('li');
    lists.forEach((li, index) => {
      li.simulate('click', fakeProps.options[index]);
    });
    expect(mockCallBack.mock.calls.length).toEqual(3);
  });
});
