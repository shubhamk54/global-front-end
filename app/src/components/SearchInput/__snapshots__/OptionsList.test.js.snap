// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for OptionsList component matches the snapshot 1`] = `
<ul
  className="suggestions"
>
  <li
    className=""
    onClick={[Function]}
  >
    test-1
  </li>
  <li
    className="suggestion-active"
    onClick={[Function]}
  >
    test-2
  </li>
  <li
    className=""
    onClick={[Function]}
  >
    test-3
  </li>
</ul>
`;
